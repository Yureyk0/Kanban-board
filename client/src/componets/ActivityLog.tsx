import { Audit } from "../types/Audit";
import {
  ChangeMessage,
  generateChangeMessages,
} from "../utils/generateChangeMessages";

interface ActivityLog {
  audit: Audit[];
  listNames: {
    value: string;
    text: string;
  }[];
}

const renderActionText = (action: ChangeMessage) => {
  const { action: actionType, oldValue, newValue, nameTask } = action;

  switch (actionType) {
    case "insert":
      return (
        <span>
          Added <b>{nameTask}</b>
        </span>
      );
    case "update":
      return (
        <span>
          Changed <b>{nameTask}</b> from <b>{oldValue as string}</b> to
          <b>{newValue as string}</b>
        </span>
      );
    case "delete":
      return (
        <span>
          Deleted <b>{nameTask}</b>
        </span>
      );
    default:
      return (
        <span>
          Unknown action for <b>{nameTask}</b>
        </span>
      );
  }
};
export const ActivityLog = ({ audit, listNames }: ActivityLog) => {
  const formatActivity = generateChangeMessages({
    audits: audit,
    listNames,
  });
  return (
    <div className="flex flex-col gap-3">
      {formatActivity.map((action, index) => (
        <div key={index} className="text-gray-500 flex justify-between text-sm">
          {renderActionText(action)}
          <p className=" whitespace-nowrap pl-2">{action.createdAt}</p>
        </div>
      ))}
    </div>
  );
};
