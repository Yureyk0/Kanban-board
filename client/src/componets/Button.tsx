export interface ButtonProps {
  onClick: () => void;
  disabled?: boolean;
  className?: string;
  type?: "submit" | "primary" | "outline";
  children: JSX.Element;
}

const Button = ({
  onClick,
  disabled = false,
  className = "",
  type = "primary",
  children,
}: ButtonProps) => {
  const buttonStyle =
    type === "submit"
      ? `bg-blue-500  text-white  ${
          disabled
            ? "bg-gray-400 cursor-not-allowed"
            : "bg-blue-500 hover:bg-blue-600 text-white"
        }`
      : type === "outline"
        ? "bg-transparent hover:bg-gray-100 py-1 px-2 border border-solid rounded font-medium"
        : "hover:bg-gray-500 py-1 px-2 border border-solid rounded bg-gray-600 text-white";

  return (
    <button
      className={`px-4 py-2 rounded focus:outline-none ${buttonStyle} ${className}`}
      onClick={onClick}
      disabled={disabled}
      type={type === "submit" ? "submit" : "button"}
    >
      {children}
    </button>
  );
};

export default Button;
