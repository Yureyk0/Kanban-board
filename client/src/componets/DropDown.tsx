import { useRef } from "react";
import { useOutsideClickHandler } from "../hooks/useOutsideClickHandler";
import DotsIcon from "../assets/dotsIcon.svg?react";
import PenIcon from "../assets/penIcon.svg?react";
import DeleteIcon from "../assets/trashIcon.svg?react";
import PlusIcon from "../assets/plusIcon.svg?react";

export interface DropDownProps {
  handleCloseDropDown: () => void;
  handleDelete: () => void;
  handleEdit: () => void;
  handleAddTask?: () => void;
  handleOpenDropDown: () => void;
  isOpen: boolean;
}

export const DropDown = ({
  handleCloseDropDown,
  handleDelete,
  handleEdit,
  handleAddTask,
  handleOpenDropDown,
  isOpen,
}: DropDownProps) => {
  const dropDownRef = useRef<HTMLDivElement>(null);

  useOutsideClickHandler(dropDownRef, handleCloseDropDown);
  return (
    <div className=" relative">
      <button
        className="px-1"
        onClick={handleOpenDropDown}
        aria-label="Open dropdown"
      >
        <DotsIcon />
      </button>
      {isOpen && (
        <div
          className="absolute right-0 top-6 w-32 bg-white rounded-md shadow-lg z-10"
          ref={dropDownRef}
          data-testid="dropdown-container"
        >
          <ul className="py-1 flex flex-col font-medium">
            <li>
              <button
                onClick={handleEdit}
                className="flex gap-2 px-2 py-2 text-sm text-gray-700 hover:bg-gray-100 w-full"
                role="Edit"
              >
                <PenIcon />
                Edit
              </button>
            </li>
            {handleAddTask && (
              <li>
                <button
                  onClick={handleAddTask}
                  className="flex gap-2 px-2 py-2 text-sm text-gray-700 hover:bg-gray-100 w-full"
                  role="Add"
                >
                  <PlusIcon /> Add task
                </button>
              </li>
            )}
            <li>
              <button
                onClick={handleDelete}
                className="flex gap-2 px-2 py-2 text-sm text-red-500 hover:bg-gray-100 w-full"
                role="Delete"
              >
                <DeleteIcon /> Delete
              </button>
            </li>
          </ul>
        </div>
      )}
    </div>
  );
};
