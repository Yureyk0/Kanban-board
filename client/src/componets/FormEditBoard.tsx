import { useState } from "react";
import { Modal } from "./Modal";
import { Board } from "../types/Board";
import { useUpdateBoardMutation } from "../app/services/taskBoardApi";

interface FormEditBoardProps {
  isOpen: boolean;
  onClose: () => void;
  board: Partial<Board>;
}

export const FormEditBoard = ({
  isOpen,
  onClose,
  board,
}: FormEditBoardProps) => {
  const [boardName, setBoardName] = useState(board.nameBoard);
  const [error, setError] = useState("");
  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
  const [updateBoard] = useUpdateBoardMutation();

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    setBoardName(value);
    setIsButtonDisabled(value.trim() === "");
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (!boardName!.trim()) {
      setError("Board Name cannot be empty");
      return;
    }
    updateBoard({ id: board.id, nameBoard: boardName });
    setBoardName("");
    setError("");
    setIsButtonDisabled(true);
    onClose();
  };

  return (
    <Modal
      isOpen={isOpen}
      onClose={() => {
        onClose();
      }}
    >
      <div className="text-center p-4">
        <h2 className="text-xl mb-4">Edit Board</h2>
        <form onSubmit={handleSubmit} className="mx-auto max-w-md">
          <div className="flex items-center mb-4">
            <label htmlFor="boardName" className="mr-2">
              Board Name:
            </label>
            <input
              id="boardName"
              type="text"
              value={boardName}
              onChange={handleChange}
              className="flex-grow px-3 py-2 border border-gray-300 rounded focus:outline-none focus:border-blue-500"
            />
          </div>
          {error && <p className="text-red-500 mb-2">{error}</p>}
          <button
            type="submit"
            className={`px-4 py-2 rounded focus:outline-none ${
              isButtonDisabled
                ? "bg-gray-400 cursor-not-allowed"
                : "bg-blue-500 hover:bg-blue-600 text-white"
            }`}
            disabled={isButtonDisabled}
          >
            Save
          </button>
        </form>
      </div>
    </Modal>
  );
};
