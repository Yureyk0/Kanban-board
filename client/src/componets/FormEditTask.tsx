import { useState } from "react";
import ReactDatePicker from "react-datepicker";
import { Task } from "../types/Task";
import { Modal } from "./Modal";

import "react-datepicker/dist/react-datepicker.css";
import { useUpdateTaskMutation } from "../app/services/taskBoardApi";
import { Select } from "./Select";

interface FormEditTaskProps {
  isOpen: boolean;
  onClose: () => void;
  task: Task;
}

interface updeteTask {
  id: string;
  nameTask: string;
  descriptionTask: string;
  priority: string;
  dueDate: Date;
}
const dataPrioritySelect = [
  { value: "Low", text: "Low" },
  { value: "Medium", text: "Medium" },
  { value: "High", text: "High" },
];

export const FormEditTask = ({ isOpen, onClose, task }: FormEditTaskProps) => {
  const { nameTask, descriptionTask, priority, dueDate, id } = task;
  const [updateTask, setUpdateTask] = useState<updeteTask>({
    id: id,
    nameTask: nameTask,
    descriptionTask: descriptionTask,
    priority: priority,
    dueDate: dueDate,
  });

  const [updateTaskQuery] = useUpdateTaskMutation();

  const handleInputChange = (
    event: React.ChangeEvent<
      HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement
    >
  ) => {
    const { name, value } = event.target;
    setUpdateTask({ ...updateTask, [name]: value });
  };

  const handleDateChange = (date: Date) => {
    setUpdateTask({ ...updateTask, dueDate: date });
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (updateTask.nameTask.trim() === "") {
      return;
    }
    updateTaskQuery(updateTask);
    onClose();
  };

  return (
    <Modal
      isOpen={isOpen}
      onClose={() => {
        onClose();
      }}
    >
      <div className="max-w-md mx-auto p-4">
        <h2 className="text-xl font-bold mb-4">Create New Task</h2>
        <form onSubmit={handleSubmit} className="space-y-4">
          <div>
            <label className="block mb-1" htmlFor="name">
              Task Name:
            </label>
            <input
              type="text"
              id="nameTask"
              name="nameTask"
              value={updateTask.nameTask}
              onChange={(event) => handleInputChange(event)}
              className="w-full px-3 py-2 border rounded-md border-gray-300 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div>
            <label className="block mb-1" htmlFor="description">
              Description:
            </label>
            <textarea
              id="descriptionTask"
              name="descriptionTask"
              value={updateTask.descriptionTask}
              onChange={(event) => handleInputChange(event)}
              className="w-full px-3 py-2 border rounded-md border-gray-300 focus:outline-none focus:border-blue-500"
              rows={3}
            />
          </div>
          <div>
            <label className="block mb-1" htmlFor="priority">
              Priority:
            </label>
            <Select
              name={"priority"}
              value={updateTask.priority}
              setValue={handleInputChange}
              data={dataPrioritySelect}
            />
          </div>
          <div>
            <label className="block mb-1" htmlFor="dueDate">
              Due Date:
            </label>
            <ReactDatePicker
              id="dueDate"
              selected={updateTask.dueDate}
              onChange={handleDateChange}
              className="w-full px-3 py-2 border rounded-md border-gray-300 focus:outline-none focus:border-blue-500"
            />
          </div>
          <div className="flex justify-end">
            <button
              type="submit"
              className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600 focus:outline-none focus:bg-blue-600 "
              disabled={updateTask.nameTask.trim() === ""}
            >
              Edit Task
            </button>
          </div>
        </form>
      </div>
    </Modal>
  );
};
