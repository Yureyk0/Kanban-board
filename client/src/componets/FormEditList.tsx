import { useState } from "react";
import { Modal } from "./Modal";
import { useUpdateListMutation } from "../app/services/taskBoardApi";
import { List } from "../types/List";
import Button from "./Button";

interface FormEditListProps {
  isOpen: boolean;
  onClose: () => void;
  list: Partial<List>;
}

export const FormEditList = ({ isOpen, onClose, list }: FormEditListProps) => {
  const [listName, setListName] = useState(list.nameList);
  const [error, setError] = useState("");
  const [isButtonDisabled, setIsButtonDisabled] = useState(true);
  const [updateList] = useUpdateListMutation();

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    setListName(value);
    setIsButtonDisabled(value.trim() === "");
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (!listName!.trim()) {
      setError("Board Name cannot be empty");
      return;
    }

    updateList({ id: list.id, nameList: listName });
    setListName(listName);
    setError("");
    setIsButtonDisabled(true);
    onClose();
  };

  return (
    <Modal
      isOpen={isOpen}
      onClose={() => {
        onClose();
      }}
    >
      <div className="text-center p-4">
        <h2 className="text-xl mb-4">Edit List</h2>
        <form onSubmit={handleSubmit} className="mx-auto max-w-md">
          <div className="flex items-center mb-4">
            <label htmlFor="listName" className="mr-2">
              List Name:
            </label>
            <input
              id="listName"
              type="text"
              value={listName}
              onChange={handleChange}
              className="flex-grow px-3 py-2 border border-gray-300 rounded focus:outline-none focus:border-blue-500"
            />
          </div>
          {error && <p className="text-red-500 mb-2">{error}</p>}
          <Button disabled={isButtonDisabled} type="submit" onClick={() => {}}>
            <span>Save</span>
          </Button>
        </form>
      </div>
    </Modal>
  );
};
