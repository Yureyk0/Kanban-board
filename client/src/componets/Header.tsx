import { useState } from "react";
import { useAddListMutation } from "../app/services/taskBoardApi";
import { SideBarHistory } from "./SideBarHistory";
import Button from "./Button";

interface HeaderProps {
  nameBoard: string;
  activeBoard: string;
  listsLength: number;
  listNames: {
    value: string;
    text: string;
  }[];
}
export const Header = ({
  nameBoard,
  activeBoard,
  listsLength,
  listNames,
}: HeaderProps) => {
  const [addList] = useAddListMutation();

  const [isSidebarOpen, setIsSidebarOpen] = useState(false);

  const toggleSidebar = () => {
    setIsSidebarOpen(!isSidebarOpen);
  };

  const handleAddList = () =>
    addList({ nameList: `New list ${listsLength}`, boardId: activeBoard });

  return (
    <header className="bg-gray-100 flex justify-between  items-center  px-4 ">
      <p className="text-2xl font-bold py-4">{nameBoard}</p>

      <div className="flex gap-3">
        <Button onClick={toggleSidebar} type="outline">
          <span>History</span>
        </Button>

        <Button onClick={handleAddList} type="primary">
          <span>+ Create new list</span>
        </Button>
        <SideBarHistory
          isOpen={isSidebarOpen}
          onClose={toggleSidebar}
          listNames={listNames}
        />
      </div>
    </header>
  );
};
