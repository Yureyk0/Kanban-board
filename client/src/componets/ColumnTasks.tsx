import { useState } from "react";
import { DropDown } from "./DropDown";
import { TaskCard } from "./TaskCard";
import { List } from "../types/List";
import {
  useAddTaskMutation,
  useDeleteListMutation,
} from "../app/services/taskBoardApi";
import { FormEditList } from "./FormEditList";
import { Task } from "../types/Task";

interface ColumnTasksProps {
  list: List;
  listNames: {
    value: string;
    text: string;
  }[];
}

const initTask: Partial<Task> = {
  nameTask: "New task",
  descriptionTask: "Description task",
  priority: "Low",
};

export const ColumnTasks = ({ list, listNames }: ColumnTasksProps) => {
  const [isOpenDropDown, setisOpenDropDown] = useState(false);
  const [isModalEditOpen, setIsModalEditOpen] = useState<boolean>(false);

  const [deleteList] = useDeleteListMutation();
  const [addTask] = useAddTaskMutation();

  const { tasks, id: listId, nameList } = list;

  const handleOpenDropDown = () => setisOpenDropDown(true);
  const handleCloseDropDown = () => setisOpenDropDown(false);

  const handleOpenModalEdit = () => setIsModalEditOpen(true);
  const handleCloseModalEdit = () => setIsModalEditOpen(false);

  const handleAddTask = () => {
    addTask({ ...initTask, listId });
    handleCloseDropDown();
  };

  const handleDeleteList = () => {
    deleteList(listId);
    handleCloseDropDown();
  };

  const handleEditList = () => {
    handleOpenModalEdit();
    handleCloseDropDown();
  };

  return (
    <div className="flex flex-col flex-shrink-0 w-64 gap-3 ">
      <div className="flex  justify-between items-center border-y-2 border-gray-300">
        <p className="py-1 font-medium">{nameList}</p>
        <div className="flex gap-3 relative">
          <span>{tasks.length}</span>
          <button className="px-1" onClick={handleOpenDropDown}>
            I
          </button>
          {isOpenDropDown && (
            <DropDown
              handleCloseDropDown={handleCloseDropDown}
              handleDelete={handleDeleteList}
              handleEdit={handleEditList}
              handleAddTask={handleAddTask}
            />
          )}
        </div>
      </div>
      <button
        onClick={handleAddTask}
        className="w-full py-1 border-2 border-dotted rounded-lg font-medium"
      >
        + Add new card
      </button>
      {tasks.length ? (
        tasks.map((task) => (
          <TaskCard
            key={task.id}
            task={task}
            listNames={listNames}
            listId={listId}
          />
        ))
      ) : (
        <div>Task list empty</div>
      )}
      <FormEditList
        isOpen={isModalEditOpen}
        onClose={handleCloseModalEdit}
        list={list}
      />
    </div>
  );
};
