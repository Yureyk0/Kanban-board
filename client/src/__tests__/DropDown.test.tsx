import { fireEvent, render, screen } from "@testing-library/react";
import { DropDown, DropDownProps } from "../componets/DropDown";

describe("DropDown Component", () => {
  const handleCloseDropDown = vi.fn();
  const handleDelete = vi.fn();
  const handleEdit = vi.fn();
  const handleAddTask = vi.fn();
  const handleOpenDropDown = vi.fn();

  const defaultProps: DropDownProps = {
    handleCloseDropDown,
    handleDelete,
    handleEdit,
    handleAddTask,
    handleOpenDropDown,
    isOpen: false,
  };

  it("renders dropdown closed by default", () => {
    const component = render(<DropDown {...defaultProps} />);
    expect(component).toMatchSnapshot();

    const dropDownContainer = screen.queryByTestId("dropdown-container");
    expect(dropDownContainer).not.toBeInTheDocument();
  });

  it("renders dropdown open when isOpen is true", () => {
    const props: DropDownProps = {
      ...defaultProps,
      isOpen: true,
    };

    render(<DropDown {...props} />);

    const dropDownContainer = screen.getByTestId("dropdown-container");
    expect(dropDownContainer).toBeInTheDocument();

    const editButton = screen.getByRole("Edit");
    const deleteButton = screen.getByRole("Delete");
    const addButton = screen.getByRole("Add");

    expect(editButton).toBeInTheDocument();
    expect(deleteButton).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
  });

  it("calls handleCloseDropDown when clicking outside dropdown", () => {
    const props: DropDownProps = {
      ...defaultProps,
      isOpen: true,
    };

    render(<DropDown {...props} />);

    fireEvent.mouseDown(document.body);

    expect(handleCloseDropDown).toHaveBeenCalledTimes(1);
  });

  it("calls handleEdit when clicking edit button", () => {
    const props: DropDownProps = {
      ...defaultProps,
      isOpen: true,
    };

    render(<DropDown {...props} />);

    const editButton = screen.getByRole("Edit");
    fireEvent.click(editButton);

    expect(handleEdit).toHaveBeenCalledTimes(1);
  });

  it("calls handleAddTask when clicking add task button", () => {
    const props: DropDownProps = {
      ...defaultProps,
      isOpen: true,
      handleAddTask: handleAddTask,
    };

    render(<DropDown {...props} />);

    const addButton = screen.getByRole("Add");
    fireEvent.click(addButton);

    expect(handleAddTask).toHaveBeenCalledTimes(1);
  });

  it("calls handleDelete when clicking delete button", () => {
    const props: DropDownProps = {
      ...defaultProps,
      isOpen: true,
    };

    render(<DropDown {...props} />);

    const deleteButton = screen.getByRole("Delete");
    fireEvent.click(deleteButton);

    expect(handleDelete).toHaveBeenCalledTimes(1);
  });
});
