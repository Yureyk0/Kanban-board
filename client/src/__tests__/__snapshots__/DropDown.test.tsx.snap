// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropDown Component > renders dropdown closed by default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" relative"
      >
        <button
          aria-label="Open dropdown"
          class="px-1"
        >
          <svg
            fill="none"
            height="800px"
            viewBox="0 0 24 24"
            width="800px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 12H12.01M12 6H12.01M12 18H12.01M13 12C13 12.5523 12.5523 13 12 13C11.4477 13 11 12.5523 11 12C11 11.4477 11.4477 11 12 11C12.5523 11 13 11.4477 13 12ZM13 18C13 18.5523 12.5523 19 12 19C11.4477 19 11 18.5523 11 18C11 17.4477 11.4477 17 12 17C12.5523 17 13 17.4477 13 18ZM13 6C13 6.55228 12.5523 7 12 7C11.4477 7 11 6.55228 11 6C11 5.44772 11.4477 5 12 5C12.5523 5 13 5.44772 13 6Z"
              stroke="#000000"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" relative"
    >
      <button
        aria-label="Open dropdown"
        class="px-1"
      >
        <svg
          fill="none"
          height="800px"
          viewBox="0 0 24 24"
          width="800px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 12H12.01M12 6H12.01M12 18H12.01M13 12C13 12.5523 12.5523 13 12 13C11.4477 13 11 12.5523 11 12C11 11.4477 11.4477 11 12 11C12.5523 11 13 11.4477 13 12ZM13 18C13 18.5523 12.5523 19 12 19C11.4477 19 11 18.5523 11 18C11 17.4477 11.4477 17 12 17C12.5523 17 13 17.4477 13 18ZM13 6C13 6.55228 12.5523 7 12 7C11.4477 7 11 6.55228 11 6C11 5.44772 11.4477 5 12 5C12.5523 5 13 5.44772 13 6Z"
            stroke="#000000"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
